pkg/kubectl/cmd/expose.go

```diff
@@ -113,11 +110,23 @@ func RunExpose(f *Factory, out io.Writer, cmd *cobra.Command, args []string) err
 		if err != nil {
 			return err
 		}
-		s, err := f.PodSelectorForResource(mapping, namespace, name)
-		if err != nil {
-			return err
+		if len(s) == 0 {
+			s, err := f.PodSelectorForResource(mapping, namespace, name)
+			if err != nil {
+				return err
+			}
+			params["selector"] = s
+		}
+		if util.GetFlagInt(cmd, "port") < 0 {
+			ports, err := f.PortsForResource(mapping, namespace, name)
+			if err != nil {
+				return err
+			}
+			if len(ports) == 0 {
+				return util.UsageError(cmd, "couldn't find a suitable port via --port flag or introspection")
+			}
+			params["port"] = ports[0]
```

Comment:
This means expose will introspect and select the port of the first container in the case of a multi-container pod. If that is the intended behavior, it should be documented in the help string, though tbh, I think you should fail here if there is more than one port, as we don't know what the user actually intends.